packages/go/stbernard/graph/transform.go

```diff
@@ -0,0 +1,346 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+package graph
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"strings"
+
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/specterops/bloodhound/dawgs"
+	"github.com/specterops/bloodhound/dawgs/drivers/pg"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/query"
+	"github.com/specterops/bloodhound/dawgs/util/size"
+	"github.com/specterops/bloodhound/graphschema"
+	"github.com/specterops/bloodhound/graphschema/common"
+	"github.com/specterops/bloodhound/packages/go/stbernard/environment"
+	"github.com/specterops/bloodhound/packages/go/stbernard/workspace"
+	"github.com/specterops/bloodhound/slicesext"
+	"github.com/specterops/bloodhound/src/migrations"
+	"github.com/specterops/bloodhound/src/model"
+	"github.com/specterops/bloodhound/src/services/graphify"
+	"github.com/specterops/bloodhound/src/services/upload"
+)
+
+const (
+	Name  = "graph"
+	Usage = "Ingest test files and transform graph data into an arrows file"
```

Comment:
Suggestion: Ingest valid collection files and transform graph data into a generic graph file